src/Development/IDE/LSP/Server.hs

```diff
@@ -16,17 +17,19 @@ import qualified Language.Haskell.LSP.Core as LSP
 import qualified Language.Haskell.LSP.Messages as LSP
 import Development.IDE.Core.Service
 
+-- | An LSP Response message can either have an error or a result.
+type ResponseBody resp = Either ResponseError resp
```

Comment:
I'd be tempted to inline this type everywhere. It's somewhat annoying to have, and not vastly shorter than the expanded version.

Reply:
I brought it in to tag the fact that this is a standard reply type. It is basically just documentation.

If you feel it should go, it can go.

Reply:
That would be the way I would have done it, but it's a reasonable choice either way.

Reply:
I prefer to just inline it. `ResponseBody` isn’t quite accurate since it also includes the “status” (in HTTP terms) of whether it has failed or not and we already have a type called `ResponseMessage` so regardless of what we chose here it will be confusing.